distribution/tools/ansi-console/src/test/java/org/elasticsearch/io/ansi/AnsiConsoleLoaderTests.java

```diff
@@ -30,35 +34,68 @@
 
     private static final AnsiOutputStream.IoRunnable NO_OP_RUNNABLE = () -> {};
 
+    private static final Charset[] charsets = new Charset[] {
+        StandardCharsets.US_ASCII,
+        StandardCharsets.ISO_8859_1,
+        StandardCharsets.UTF_8,
+        StandardCharsets.UTF_16,
+        StandardCharsets.UTF_16LE,
+        StandardCharsets.UTF_16BE };
+
     public void testNullOutputIsNotConsole() {
-        assertThat(AnsiConsoleLoader.isValidConsole(null), is(false));
+        assertThat(AnsiConsoleLoader.newConsole(null), nullValue());
     }
 
     public void testRedirectedOutputIsNotConsole() {
         try (AnsiPrintStream ansiPrintStream = buildStream(AnsiType.Redirected, randomIntBetween(80, 120))) {
-            assertThat(AnsiConsoleLoader.isValidConsole(ansiPrintStream), is(false));
+            assertThat(AnsiConsoleLoader.newConsole(ansiPrintStream), nullValue());
         }
     }
 
     public void testUnsupportedTerminalIsNotConsole() {
         try (AnsiPrintStream ansiPrintStream = buildStream(AnsiType.Unsupported, randomIntBetween(80, 120))) {
-            assertThat(AnsiConsoleLoader.isValidConsole(ansiPrintStream), is(false));
+            assertThat(AnsiConsoleLoader.newConsole(ansiPrintStream), nullValue());
         }
     }
 
     public void testZeroWidthTerminalIsNotConsole() {
         try (AnsiPrintStream ansiPrintStream = buildStream(randomFrom(SUPPORTED_TERMINAL_TYPES), 0)) {
-            assertThat(AnsiConsoleLoader.isValidConsole(ansiPrintStream), is(false));
+            assertThat(AnsiConsoleLoader.newConsole(ansiPrintStream), nullValue());
         }
     }
 
     public void testStandardTerminalIsConsole() {
+        int width = randomIntBetween(40, 260);
+        try (AnsiPrintStream ansiPrintStream = buildStream(randomFrom(SUPPORTED_TERMINAL_TYPES), width)) {
+            ConsoleLoader.Console console = AnsiConsoleLoader.newConsole(ansiPrintStream);
+            assertThat(console, notNullValue());
+            assertThat(console.width().get(), is(width));
+        }
+    }
+
```

Comment:
Is there a test we can write that would fail if `tryExtractPrintCharset` fails with a reflection related error ? Could be useful for upgrades

Reply:
Hmm, I don't have good ideas. The method would fail if the classes in the JANSI lib change structure, following an upgrade, but we can't simulate that in a test.

We do not have tests asserting the auto-configuration message to the terminal.
If we would, we could assert the unicode formatting on at least some platforms.

Reply:
We assert the message to the terminal in ArchiveGenerateInitialCredentialsTests for some simple things but we need to adjust and fix these tests ( currently muted ) and we can add additional cases for sure

Reply:
Good, I missed those!

Reply:
TBC,when those tests are un-muted, I believe we can assert there that the displayed information contains unicode chars (on some platforms, eg not Windows), which also implies that reflection worked.